### PR TITLE
Add Agent navigation

### PR DESCRIPTION
**Pull request by rubene**
🐘  🎨 
* Because we are using the layout gem for the POS and the user is
  significantly different from the member/non-member create a separate
  views for the header and sidebar menus navigation. This follow the
  same flow reading menu items from .yml file and creating the
  navigation.
* Create this branch from the rails-5 branch because is the one being in
  use by POS.

VSTS: https://amaabca.visualstudio.com/driver_education_backlog/_workitems?id=1503&_a=edit

### PR Review Checklist
To be completed by the person who opens the PR. Use strikethroughs for items that are not applicable in list.
- [x] I have reviewed my own PR to check syntax & logic, removed unnecessary/old code, made sure everything aligns with our style guide and BEM.
~- [] I've added new components to the style guide (or have a PR in to add them).~
- [x] Any applicable version numbers have been updated.
- [] I've tested on mobile, tablet, and desktop, as well as across all of the browsers we support.
- [x] I've proof-read all text for legibility, proper grammar, punctuation, and capitalization (titlecase).
- [x] I have written a detailed PR message explaining what is being changed and why
- [x] I’ve linked to any relevant VSTS tickets
- [x] I’ve added the appropriate review symbols to my PR - (elephant) for dev review, (art) for design

screen shot 2017-08-23 at 9 33 12 am

screen shot 2017-08-23 at 9 30 51 am

### CODE REVIEW

**Comment by MathieuGilbert** [comment]
📖

**Comment by MathieuGilbert** [comment]
@rubene Could you add some screenshots showing this in use?

File: app/views/ama_layout/agent/_siteheader.html.erb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<div class="header">
+  <div class="title-bar" data-responsive-toggle="main-menu" data-hide-for="large">
+    <button class="menu-icon" type="button" data-toggle="offCanvasLeft"></button>
+    <%= link_to(image_tag("ama-logo.png"), "https://www.ama.ab.ca", class: "title-bar__logo") %>
```
Comment (MathieuGilbert) [inline app/views/ama_layout/agent/_siteheader.html.erb]: Is there a config variable for the WWW URL you can use? I see the other layout hard-codes it to prod too anyways.

File: lib/ama_layout/decorators/agent/navigation_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+module AmaLayout
+  module Agent
+    class NavigationDecorator < Draper::Decorator
+      delegate_all
+
+      def items
+        object.items.map { |i| i.decorate }
```
Comment (MathieuGilbert) [inline lib/ama_layout/decorators/agent/navigation_decorator.rb]: You can use `.map(&:decorate)` here.

File: spec/agent/navigation_spec.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    end
+  end
+
+  describe "#items" do
+    it "does not remove nil #navigation_items" do
+      expect(subject.items.collect {|i| i.alt }).to include nil
+    end
+
+    it "displays links " do
+      expect(subject.items.collect {|i| i.link }).to eq subject.navigation_items.collect {|i| i["link"] }
```
Comment (MathieuGilbert) [inline spec/agent/navigation_spec.rb]: Could use `.collect(&:link)`, but I don't think the `i["link"]` part has a shortcut like that.

**Comment by MathieuGilbert** [comment]
:poop:

**Comment by DaveAMA** [comment]
📖

**Comment by rubene** [comment]
:wave:

**Comment by MathieuGilbert** [comment]
:elephant: :elephant: :elephant:

**Pull request merged**